controllers/psyProfileController.ts

```diff
@@ -103,8 +106,50 @@ const editPsyProfile = async (req: Request, res: Response): Promise<void> => {
   });
 };
 
+const activate = async (req: Request, res: Response): Promise<void> => {
+  await dbPsychologists.activate(req.params.psyId);
+
+  res.json({
+    success: true,
+    message: 'Vos informations sont de nouveaux visibles sur l\'annuaire.',
+  });
+};
+
+const suspendValidators = [
+  check('date')
+    .isISO8601()
+    .isAfter()
+    .withMessage('Vous devez spécifier une date de fin de suspension dans le futur.'),
+  check('reason')
+    .trim().not().isEmpty()
+    .customSanitizer((value, { req }) => req.sanitize(value))
+    .withMessage('Vous devez spécifier une raison.'),
+];
+
+const suspend = async (req: Request, res: Response): Promise<void> => {
+  validation.checkErrors(req);
+
+  // TO REMOVE suspensionDepartment
+  if (config.suspensionDepartment) {
+    const psy = await dbPsychologists.getPsychologistById(req.params.psyId);
+    if (!config.suspensionDepartment.includes(psy.departement)) {
+      throw new CustomError('Feature non disponible pour le moment', 400);
```

Comment:
Shouldn't this be checked before the field validation? Maybe as a middleware?

Reply:
no, because we need the department of the psychologist :/